index.html

```diff
@@ -1583,19 +1585,19 @@ <h2>The <dfn>ExposedThing</dfn> interface</h2>
         Promise&lt;void&gt; destroy();
       };
 
-      callback PropertyReadHandler = Promise&lt;InteractionData&gt;(
+      callback PropertyReadHandler = Promise&lt;any&gt;(
               optional InteractionOptions options = null);
 
-      callback PropertyWriteHandler = Promise&lt;void&gt;(any value,
+      callback PropertyWriteHandler = Promise&lt;void&gt;(InteractionData value,
               optional InteractionOptions options = null);
 
-      callback ActionHandler = Promise&lt;InteractionData&gt;(any params,
+      callback ActionHandler = Promise&lt;any&gt;(any params,
```

Comment:
So following this argument:

>Done. Actually it's the exact opposite of how we do it in ConsumedThing.
When data comes from the implementation, it's InteractionData, when it goes there, it's any.

Shouldn't params be `InteractionData`?

Reply:
Now that is tricky. In ConsumedThing the params can be a single value, array or an object (used for multiple parameters).

Here in an action handler we may get a single value, array, `InteractionData`, or an object with {simple values, array, `InteractionData` or other objects}, recursively.

This part is simply underspecified - please make suggestions how to handle it. Maybe we can fix that in a separate PR as well.

If we use `InteractionData`, all of the parsing falls on the app, which may be OK, but that functionality is already implemented in the ConsumedThing implementation underneath, so we'd force the app to do something that's also done by the impl - in which case why should an app bother using a WoT Scripting implementation, just make a shim over a comms lib.

If we use `any`, perhaps we should make a note about what to expect, and specify an algorithm on what to do when an action is invoked - it will be similar to the write steps. This seems to be the more developer-friendly option, so I'd prefer this. 

The TODO is then not changing the signature, but completing the algorithm on action requests, if you agree.

Reply:
> Now that is tricky. In ConsumedThing the params can be a single value, array or an object (used for multiple parameters).

Maybe I miss this: How can we specify different parameters on the TD? As far as I know, we have only the `input` property to specify the Dataschema of the input. In this sense, I do not see any difference with the `writeHandler`.

Reply:
The TD spec has always had trouble with multiplicity :). The idea was that one `input` is enough since it can describe multiple inputs as an object, and it can also describe a single param just as a single param. It would have been more clear if `input` was an array of DataSchema's, but it's this way now. :)

Reply:
Though I admit that representing multiple params as an object is better since if all of them are optional, you can name the ones used, instead of filling the sig with nulls. A bit awkward, perhaps not well explained in the TD spec, but works.

Reply:
So should we move this discussion in a new issue/PR? it is fine to leave the PR as it is and change it later.

Reply:
No, I will fix the algorithm a bit later, maybe on Friday. We'll need to review it again it seems. 
About multiple properties handling, it is underspecified now (it's implementation specific how do they tell it's a multiple properties request and how they deserialize the property names). We can handle _that_ in another PR.
Anyway, it's better to comment in the PR than on the commits directly, since this discussion will be buried :).

Reply:
Looks like after checking the TD, it is possible to use `InteractionData` when passing args to the handler, which I did.